### DISCUSSION TITLE
How to print the name of a file uploaded via a button?

### DISCUSSION THREAD

**Discussion by hopezh** [marimo-team/marimo#3256] [Q&A] [answered]
As shown below, I have two questions:

1. how to access the `value` of the file updated via the button created in the first cell? 

2. Why there's no button for file uploading after running the second cell?

![image](https://github.com/user-attachments/assets/10d43b71-819d-40e4-aac8-2405434f38d4)

**Comment by akshayka** [comment, marked as answer]
@hopezh , you have to assign UI elements to global variables and output the global variable. Your displayed button doesn't have an associated variable so it doesn't work.

You also need to end the cell with `f` in order to display the button.

Try this instead: https://github.com/marimo-team/marimo/blob/main/examples/ui/file.py

Tip: You can run this notebook with

```
marimo edit https://github.com/marimo-team/marimo/blob/main/examples/ui/file.py
```

at the command line.

![image](https://github.com/user-attachments/assets/1d20eb54-0bfc-4639-b809-f79890a512c5)

For many more examples, see: https://github.com/marimo-team/marimo/tree/main/examples

For a tutorial on using UI widgets please run: `marimo tutorial ui` at the command-line.
